itkwidgets/integrations/__init__.py

```diff
@@ -1,60 +1,132 @@
-            return vtk_image_to_ndarray(image)
-    if HAVE_DASK:
-        import dask
+            spatial_image = vtk_image_to_spatial_image(image)
+            scale_factors = _spatial_image_scale_factors(spatial_image, min_length)
+            multiscale = to_multiscale(spatial_image, scale_factors)
+    if multiscale is None:
         if isinstance(image, dask.array.core.Array):
-            return dask_array_to_ndarray(image)
-    if HAVE_TORCH:
+            spatial_image = to_spatial_image(image)
+            scale_factors = _spatial_image_scale_factors(spatial_image, min_length)
+            multiscale = to_multiscale(spatial_image, scale_factors)
+    if multiscale is None:
+        if isinstance(image, zarr.Array):
+            spatial_image = to_spatial_image(image)
+            scale_factors = _spatial_image_scale_factors(spatial_image, min_length)
+            multiscale = to_multiscale(spatial_image, scale_factors)
+    if multiscale is None:
+        # NGFF Zarr
+        if isinstance(image, zarr.Group) and 'multiscales' in image.attrs:
+            return image.store
+    if multiscale is None and HAVE_TORCH:
         import torch
         if isinstance(image, torch.Tensor):
-            return image.numpy()
-    if HAVE_XARRAY:
-        import xarray
-        if isinstance(image, xarray.DataArray):
+            spatial_image = to_spatial_image(image.numpy())
+            scale_factors = _spatial_image_scale_factors(spatial_image, min_length)
+            multiscale = to_multiscale(spatial_image, scale_factors)
+    if multiscale is None:
+        # Todo: preserve dask Array, if present, check if dims are NGFF -> use dims, coords
+        # Check if coords are uniform, if not, resample
+        if isinstance(image, xr.DataArray):
             return xarray_data_array_to_numpy(image)
-        if isinstance(image, xarray.Dataset):
+        if isinstance(image, xr.Dataset):
             return xarray_data_set_to_numpy(image)
-    return image
+    if multiscale is None:
+        if isinstance(image, np.ndarray):
+            spatial_image = to_spatial_image(image)
+            scale_factors = _spatial_image_scale_factors(spatial_image, min_length)
+            multiscale = to_multiscale(spatial_image, scale_factors)
+    if multiscale is None:
+        raise RuntimeError("Could not process the viewer image")
+
+    # Todo: for very large images serialize to disk cache
+    store = zarr.storage.MemoryStore(dimension_separator='/')
+    multiscale.to_zarr(store, compute=True)
+    return store
 
 
 def _get_viewer_point_sets(point_sets):
     if HAVE_VTK:
         import vtk
         if isinstance(point_sets, vtk.vtkPolyData):
             return vtk_polydata_to_vtkjs(point_sets)
-    if HAVE_DASK:
-        import dask
-        if isinstance(point_sets, dask.array.core.Array):
-            return dask_array_to_ndarray(point_sets)
+    if isinstance(point_sets, dask.array.core.Array):
+        return dask_array_to_ndarray(point_sets)
```

Comment:
It looks like `dask_array_to_ndarray` has been removed. Should we keep it around for this case or just move the functionality here?

Reply:
Good catch. Let's just use `np.asarray` here for now.